### PR TITLE
arm: support arm build

### PR DESCRIPTION
**Pull request by rootfs**
Generated the vmlinux.h from an ARM server via
```
bpftool btf dump file /sys/kernel/btf/vmlinux format c > vmlinux.h
```

Tested on Ampere setup:
```
[root@perf-arm-12 kepler]# uname -r
5.14.0-362.21.1.el9_3.aarch64
[root@perf-arm-12 kepler]# curl localhost:8888/metrics 2>&1 |grep kepler_exporter_build_info
# HELP kepler_exporter_build_info A metric with a constant '1' value labeled by version, revision, branch, goversion from which kepler_exporter was built, and the goos and goarch for the build.
# TYPE kepler_exporter_build_info gauge
kepler_exporter_build_info{branch="",goarch="arm64",goos="linux",goversion="go1.20.14",revision="unknown",tags="include_gcs,include_oss,containers_image_openpgp,gssapi,providerless,netgo,osusergo,libbpf,linux,gpu",version=""} 1
[root@perf-arm-12 kepler]# curl localhost:8888/metrics |grep kepler_process_ |grep joules |sort -k 2 -g |tail
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  132k    0  132k    0     0  32.4M      0 --:--:-- --:--:-- --:--:-- 32.4M
kepler_process_core_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="86862",source="intel_rapl",vm_id=""} 0.195
kepler_process_package_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="86862",source="intel_rapl",vm_id=""} 0.195
kepler_process_core_joules_total{command="sh",container_id="system_processes",mode="dynamic",pid="86735",source="intel_rapl",vm_id=""} 0.549
kepler_process_package_joules_total{command="sh",container_id="system_processes",mode="dynamic",pid="86735",source="intel_rapl",vm_id=""} 0.549
kepler_process_core_joules_total{command="irqbalance",container_id="system_processes",mode="dynamic",pid="2403",source="intel_rapl",vm_id=""} 0.681
kepler_process_package_joules_total{command="irqbalance",container_id="system_processes",mode="dynamic",pid="2403",source="intel_rapl",vm_id=""} 0.681
kepler_process_core_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="87516",source="intel_rapl",vm_id=""} 1.581
kepler_process_package_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="87516",source="intel_rapl",vm_id=""} 1.581
kepler_process_core_joules_total{command="kworker/73:0",container_id="kernel_processes",mode="dynamic",pid="383",source="intel_rapl",vm_id=""} 229.143
kepler_process_package_joules_total{command="kworker/73:0",container_id="kernel_processes",mode="dynamic",pid="383",source="intel_rapl",vm_id=""} 229.143
```

### CODE REVIEW

**Comment by rootfs** [comment]
@vimalk78 @SamYuan1990 how did you build the arm64 kepler_builder image? I can find one for s390 but not arm64
image

**Comment by SamYuan1990** [comment]
checking

**Comment by vimalk78** [comment]
> @vimalk78 @SamYuan1990 how did you build the arm64 kepler_builder image? I can find one for s390 but not arm64

I built it on internal RH systems, not in upstram CI.

**Comment by SamYuan1990** [comment]
let see what will happen in #1261

File: build/Dockerfile [hunk 1]
```diff
@@ -25,8 +26,10 @@ RUN INSTALL_PKGS=" \
 	yum install -y $INSTALL_PKGS 
 
 RUN if [[ ! -z "$INSTALL_DCGM" ]]; then \
-    dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo; \
-    yum install -y datacenter-gpu-manager; \
+        if [ "$TARGETARCH" == "amd64" ]; then \
```
Comment (rootfs) [inline build/Dockerfile:29]: dcgm is only available on x86

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -60,14 +60,3 @@ jobs:
           src:
             - './.github/workflows/image_base.yml'
             - './build/Dockerfile.builder'
-
-  image_base:
```
Comment (rootfs) [inline .github/workflows/pull_request.yml:64]: @SamYuan1990 I removed this due to the CI error [here](https://github.com/sustainable-computing-io/kepler/actions/runs/8054337624)

**Comment by rootfs** [comment]
Tested it on ARM
```console
# curl localhost:8888/metrics |grep joules |sort -k 2 -g |grep kepler_process_core |tail -5
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  131k    0  131k    0     0  21.3M      0 --:--:-- --:--:-- --:--:-- 21.3M
kepler_process_core_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="268421",source="intel_rapl",vm_id=""} 0.033
kepler_process_core_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="256051",source="intel_rapl",vm_id=""} 0.06
kepler_process_core_joules_total{command="node",container_id="system_processes",mode="dynamic",pid="256093",source="intel_rapl",vm_id=""} 0.075
kepler_process_core_joules_total{command="irqbalance",container_id="system_processes",mode="dynamic",pid="2403",source="intel_rapl",vm_id=""} 0.3
kepler_process_core_joules_total{command="migration/71",container_id="kernel_processes",mode="dynamic",pid="371",source="intel_rapl",vm_id=""} 104.277
# rpm -qa |grep kepler
kepler-v0.7.7.2.gf4b2922d.dirty-20240226.aarch64
# uname -r
5.14.0-362.21.1.el9_3.aarch64
```

**Review by vprashar2929 — APPROVED** [review]
LGTM

**Review by SamYuan1990 — CHANGES_REQUESTED** [review]
wait for CI and tooling scope discussion.

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (SamYuan1990) [reply .github/workflows/pull_request.yml:64]: https://github.com/sustainable-computing-io/kepler/pull/1262#pullrequestreview-1903252884 yes... @rootfs you had disabled workflow call for the job. hence we need to add workflow call back.

Comment (SamYuan1990) [reply .github/workflows/pull_request.yml:64]: we need to revert or roll back https://github.com/sustainable-computing-io/kepler/pull/1262 as this breaks lots of CI pipelines.

Comment (SamYuan1990) [reply .github/workflows/pull_request.yml:64]: btw, #1262 https://github.com/sustainable-computing-io/kepler/actions/runs/8051024777 PR merged with CI failure?

Comment (SamYuan1990) [reply .github/workflows/pull_request.yml:64]: @rootfs , please check https://github.com/sustainable-computing-io/kepler/pull/1267

File: Makefile [hunk 3]
```diff
@@ -154,7 +154,6 @@ container_build:
 		git config --global --add safe.directory /kepler && make build
 
 build_rpm:
-	rpmautospec generate-changelog packaging/rpm/kepler.spec
```
Comment (SamYuan1990) [inline Makefile:157]: if we remove this, will it fails on RPM build?

Comment (SamYuan1990) [reply Makefile:157]: if we don't need rpmautospec, hence https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile.builder#L33-L34 those lines should be removed as well?

File: .github/workflows/pull_request.yml (hunk 2 above)
Comment (rootfs) [reply .github/workflows/pull_request.yml:64]: that makes sense, i merged 1267

File: Makefile (hunk 3 above)
Comment (rootfs) [reply Makefile:157]: i copied CHANGELOG.md to rpm change log

Comment (rootfs) [reply Makefile:157]: > if we don't need rpmautospec, hence https://github.com/sustainable-computing-io/kepler/blob/main/build/Dockerfile.builder#L33-L34 those lines should be removed as well?

good catch, removed

**Review by SamYuan1990 — APPROVED** [review]
LGTM

**Comment by rootfs** [comment]
kepler amd64 and arm64 images are now available
image

**Comment by rootfs** [comment]
> Awesome! Works as expected 👍

@beneiltis this sounds great! would you please write a PR and share your story? https://github.com/sustainable-computing-io/kepler-doc

**Comment by beneiltis** [comment]
Sure I am happy to help. But can you elaborate on what I should create a PR for? 😇 
I only checked that the image works on arm64 clusters.

**Comment by rootfs** [comment]
> Sure I am happy to help. But can you elaborate on what I should create a PR for? 😇 I only checked that the image works on arm64 clusters.

@beneiltis The installation process on ARM is not there yet. It would be great if you can walk through your steps and share how to get started. https://github.com/sustainable-computing-io/kepler-doc/blob/main/docs/installation/kepler.md

**Comment by Maarham** [comment]
I had a clarification regarding this implementation. The implementation has "source=intel_rapl". Is that a display artifact and how does it work with ARM64 when you are using the intel_rapl? Thanks!

**Pull request merged**